server/index.js

```diff
@@ -95,34 +95,59 @@ app.get('/api/games/retrieve/:userId', (req, res, next) => {
   }
 
   const sql = `
-    select "users"."username"
+    select "users"."username",
+           "games"."state",
+           "games"."gameId"
       from "games"
 inner join "users"
         on "challenger" = "userId"
         or "opponent" = "userId"
         where "challenger" = $1
-        or "opponent" = $1;
+        or "opponent" = $1
+        limit 1;
   `;
 
   const params = [userId];
   db.query(sql, params)
     .then(result => {
-      if (!result.rows[0].username || !result.rows[1].username) {
+      if (!result.rows[0].username) {
         throw new ClientError(400, 'this user is not in any active games');
       }
       res.status(200).json(result.rows);
     })
     .catch(err => next(err));
 });
 
+app.patch('/api/games/:gameId', (req, res, next) => {
+  const { gameId } = req.params;
+  const state = req.body;
+  const sql = `
+    update "games"
+       set "state" = $2
+       where "gameId" = $1
+       returning "state"
+  `;
+  const params = [gameId, state];
+
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        throw new ClientError(400, 'this gameId does not exist');
+      }
+      res.status(200).json(result.rows[0].state);
+    })
```

Comment:
If you include the room id in the request body, you can broadcast this update back to the clients

```js
io.to(roomId).emit('flip-card', updatedState)
```

Then in your client code, you can listen for `'flip-card'` and update both clients at the same time.

Reply:
cha chinggggg